index.js

```diff
@@ -14,13 +14,14 @@ export default function mergeStreams(streams) {
 	const objectMode = streams.some(({readableObjectMode}) => readableObjectMode);
 	const highWaterMark = getHighWaterMark(streams, objectMode);
 	const passThroughStream = new PassThroughStream({objectMode, writableHighWaterMark: highWaterMark, readableHighWaterMark: highWaterMark});
-	passThroughStream.setMaxListeners(Number.POSITIVE_INFINITY);
 
 	if (streams.length === 0) {
 		passThroughStream.end();
 		return passThroughStream;
 	}
 
+	passThroughStream.setMaxListeners(passThroughStream.getMaxListeners() + streams.length);
```

Comment:
> It also sets 2 end event listeners and 1 data event listener on each input stream.

Wouldn't that be 3 events per stream?

Reply:
The `end` and `data` event listeners are set on each input stream, not on the merged stream.
The input streams do not have an issue with maxListeners, since they receive the same amount of event listeners regardless of how many other input streams are passed as argument.

The maxListeners is per eventEmitter + event name, not only per eventEmitter.
The merged stream receives a single listener for each event ([link to source in Node.js](https://github.com/nodejs/node/blob/0f461aad4cbf747fa053714f08e5030ddf019977/lib/internal/streams/readable.js#L907-L1087)).

Reply:
Thanks for explaining 👍